src/TransferList/TransferList.types.ts

```diff
@@ -1,14 +1,79 @@
-export type TransferListProps = {
+export type TransferListItem =
+  | string
+  | { primaryLabel: string; secondaryLabel?: string; id: string };
```

Comment:
This means that typescript is entirely happy to let me have a mixed type list but incorrectly types the callback:

![image](https://github.com/user-attachments/assets/8945a315-b5bf-496e-8c09-a9c07f60252e)


![image](https://github.com/user-attachments/assets/30ab5dcf-f90b-427f-a789-77418982bc5b)